internal/provider/job_resource.go

```diff
@@ -267,7 +267,10 @@ func (r *JobResourceModel) ParseHttpResponse(body []byte) diag.Diagnostics {
 	r.URL = types.StringValue(resultApiJob.URL)
 	r.Status = types.StringValue(resultApiJob.Status)
 	r.TemplateID = types.Int64Value(resultApiJob.TemplateID)
-	r.InventoryID = types.Int64Value(resultApiJob.Inventory)
+	if r.InventoryID.ValueInt64() == 0 {
+		r.InventoryID = types.Int64Value(resultApiJob.Inventory)
+	}
```

Comment:
I feel like the bug this is trying to fix was introduced by https://github.com/ansible/terraform-provider-aap/pull/10/files#diff-afd9b829236927b85ef1814941bb6f012eed9cca71547f9f9ce15dcceb29a6f4R216-R219. I don't know why that change was made, but, among other things, it makes it impossible to use a job template that has a default inventory and is configured not to prompt on launch. I think fixing that bug makes this bug go away, because it shouldn't be possible to launch a job with prompt on launch and no default inventory set.